### PR TITLE
bugfix: reset mutable state

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

File: service/history/conflictResolver.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ func (r *conflictResolver) reset(replayEventID int64, startTime time.Time) (*mut
 	resetMutableStateBuilder.executionInfo.StartTimestamp = startTime
 	// the last updated time is not important here, since this should be updated with event time afterwards
 	resetMutableStateBuilder.executionInfo.LastUpdatedTimestamp = startTime
+	resetMutableStateBuilder.updateReplicationStateLastEventID(sourceCluster, replayEventID)
```
Comment (samarabbas) [inline service/history/conflictResolver.go]: I think just setting the lastWriteEventID won't be enough.  We also need to set the lastWriteVersion correctly too.

File: service/history/historyReplicator.go [hunk 2]
```diff
@@ -142,7 +142,7 @@ func (r *historyReplicator) ApplyEvents(request *h.ReplicateEventsRequest) (retE
 					request.GetSourceCluster(), request.GetVersion(), request.GetFirstEventId(), request.GetNextEventId())
 
 				resolver := newConflictResolver(r.shard, context, r.historyMgr, r.logger)
-				msBuilder, err = resolver.reset(ri.GetLastEventId(), msBuilder.executionInfo.StartTimestamp)
+				msBuilder, err = resolver.reset(request.GetSourceCluster(), ri.GetLastEventId(), msBuilder.executionInfo.StartTimestamp)
```
Comment (samarabbas) [inline service/history/historyReplicator.go]: I think it is incorrect to pass in the source cluster on the replication task.  It needs to be inferred from the lastWriteVersion.

File: service/history/conflictResolver.go (hunk 1 above)
Comment (wxing1292) [reply service/history/conflictResolver.go]: those are updated in #806

File: service/history/historyReplicator.go (hunk 2 above)
Comment (wxing1292) [reply service/history/historyReplicator.go]: those are updated in #806

**Comment by wxing1292** [comment]
close in favor of #806

**Pull request closed without merging**